include/boost/math/special_functions/fibonacci.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020, Madhur Chauhan
+
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_SPECIAL_FIBO_HPP
+#define BOOST_MATH_SPECIAL_FIBO_HPP
+
+#include <boost/math/constants/constants.hpp>
+#include <boost/math/policies/error_handling.hpp>
+#include <cmath>
+#include <limits>
+
+#ifdef _MSC_VER
+#pragma once
+#endif
+
+namespace boost {
+namespace math {
+namespace detail {
+// this should be constexpr in future
```

Comment:
Have you checked godbolt to see if these get compiled into constants? If not, we might want to put:

```
fib_bits_phi = 0.6942419136306174; // = ln(phi)/ln(2)
```

Reply:
[Yes, they are reduced into constants](https://compiler-explorer.com/z/4a6dGP)